tidy3d/material_library/material_library.py

```diff
@@ -80,6 +82,80 @@ def medium(self):
         return self.variants[self.default].medium
 
 
+class Variant2D(Tidy3dBaseModel):
+    """Reference, data_source, and material model for a variant of a 2D material."""
+
+    medium: Medium2D = pd.Field(
+        ...,
+        title="Material dispersion model",
+        description="A dispersive medium described by the pole-residue pair model.",
+    )
+
+    medium_parametric: Callable = pd.Field(
+        None,
+        title="Parametric material model",
+        description="Parameteric 2d material dispersion model.",
```

Comment:
can you be more specific about what this is? eg. a function of ... that returns ...? How does this field affect the medium?
Note also that for `Callable` it is best to put the arg types and return type, eg.
```py
Callable[[float, str], int]
```
see [here](https://docs.python.org/3/library/typing.html#typing.Callable).

Reply:
I want to allow for a variable number of parameters though, so we don't know the argument types in advance.